src/iris/api.py

```diff
@@ -6219,12 +6221,22 @@ class SenderHeartbeat():
     internal_allowlist_only = True
 
     def __init__(self, config):
-        cfg = config.get("iris-message-processor", {})
-        self.zk_hosts = cfg.get("zk_hosts", "127.0.0.1:2181")
+        cfg = config.get("external_sender", {})
         self.lock_path = cfg.get("lock_path", "/iris_message_processor/db_locks/bucket_lock")
         self.number_of_buckets = cfg.get("number_of_buckets", 100)
-        self.sender_ttl = config.get("sender_ttl", 60)
-        self.zk_debug = config.get("zk_debug", True)
+        self.sender_ttl = cfg.get("sender_ttl", 60)
+        self.zk_debug = cfg.get("zk_debug", True)
+        if not self.zk_debug:
```

Comment:
This means when zk_debug=True, we won't start kazoo client. Is it expected?

Reply:
yeah, this option means within SenderHeartbeat() exitstack will to replace zookeeper lock context manager for testing so there is no need for a kazoo client to be started if this option is enabled